feat(pie-checkbox): DSW-1574 checkbox aria property (#1440)
justeattakeaway/pie: packages/components/pie-checkbox/README.md

| Property | Type | Default | Description |
|---|---|---|---|
| `name` | `string` | - | The name of the checkbox (used as a key/value pair with `value`). This is required in order to work properly with forms. |
| `value` | `string` | 'on' | The value of the input (used as a key/value pair in HTML forms with `name`). If not passed falls back to the html default value "on". |
| `required` | `boolean` | `false` | If true, the checkbox is required to be checked before submitting the form. If it is not in checked state, the component validity state will be invalid. |

> | `label` | `string` | '' | Text associated with the checkbox. If not passed make sure to either pass label or labeledby/describedby to aria object. |

Reviewer: ```suggestion
| `label` | `string` | '' | Text associated with the checkbox. If not passed make sure to either pass label or labeledby/describedby to the aria prop. |
```

> | `label` | `string` | '' | Text associated with the checkbox. If not passed make sure to either pass label or labeledby/describedby to aria object. |

Reviewer 2: ```suggestion
| `label` | `string` | '' | Text associated with the checkbox. If there is no label to provide, make sure to either pass label, labelledby or describedby to the aria property. |
```
